### DISCUSSION TITLE
Unable to use shapes (hex and arrowHead) in Continuous canvas

### DISCUSSION THREAD

**Discussion by ghost** [mesa/mesa#1936] [Q&A] [answered]
I am trying to use shapes other than circle and rect for some of my agents but it is not working in continuous canvas. Below is my code snip for portrayal:
<img width="806" alt="image" src="https://github.com/projectmesa/mesa/assets/131771458/341b5716-dc5c-4091-b747-66b939c7804e">

I am not sure if I followed the examples "hex_snowflake" and "shape_example" correctly. But I did notice that both these use a grid  but I am using a continuous canvas. Is this the issue?
Thanks for the help!

**Comment by rht** [comment, marked as answer]
Sorry for the late response. The shape is specific to the `Grid`, not available in `ContinuousSpace`. If you want custom marker for the `ContinuousSpace`, you may use the experimental Solara-based visualization. Here is an example of using Matplotlib to draw the grid on virus on network: https://github.com/projectmesa/mesa-examples/blob/main/examples/virus_on_network/app.py. You could use the usual Matplotlib scatter plot with any marker of your choice.

**Reply by ghost** [reply]
Thank you for the information. I will look at the examples!

**Reply by rht** [reply]
https://github.com/projectmesa/mesa/blob/44710477e9cadeb05efb9bff06541efc596b3d35/mesa/experimental/jupyter_viz.py#L380-L402 has code for continuous space.

**Reply by ghost** [reply]
Thank you :)
